frontends/p4/simplifyDefUse.cpp

```diff
@@ -339,6 +574,83 @@ class FindUninitialized : public Inspector {
         return loc;
     }
 
+    // Used in header-to-header and struct-to-struct assignments
+    void processHeadersInAssignment(const StorageLocation* dst, const StorageLocation* src) {
+        if (!dst || !src)
+            return;
+
+        auto dst_struct_storage = dst->to<StructLocation>();
+        auto src_struct_storage = src->to<StructLocation>();
+
+        if (!dst_struct_storage || !src_struct_storage)
+            return;
+
+        if (dst_struct_storage->isHeader() && src_struct_storage->isHeader()) {
+            auto val = headerDefs->find(src);
+            if (val)
+                headerDefs->update(dst, *val);
+            else
+                headerDefs->remove(dst);
+
+            return;
+        }
+
+        if (dst_struct_storage->isStruct() && src_struct_storage->isStruct()) {
+            auto dst_fields = dst_struct_storage->fields();
+            auto src_fields = src_struct_storage->fields();
+
+            auto it1 = dst_fields.begin();
+            auto it2 = src_fields.begin();
+            while (it1 != dst_fields.end() && it2 != src_fields.end()) {
+                processHeadersInAssignment(*it1, *it2);
+                ++it1;
+                ++it2;
+            }
+        }
+    }
```

Comment:
BUG_CHECK to make sure that no other cases exist?

Reply:
That's a great idea!